hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieLogFileReader.java

```diff
@@ -210,15 +214,33 @@ private HoodieLogBlock readBlock() throws IOException {
   private HoodieLogBlock createCorruptBlock() throws IOException {
     log.info("Log " + logFile + " has a corrupted block at " + inputStream.getPos());
     long currentPos = inputStream.getPos();
-    long nextBlockOffset = scanForNextAvailableBlockOffset();
+    byte[] corruptedBytes;
+    int corruptedBlockSize = 0;
+    boolean done = true;
+    BufferedInputStream bufferedInputStream = new BufferedInputStream(inputStream);
+    do {
+      // read upto logblocksize to find next magic header
+      corruptedBytes = new byte[logBlockSize];
+      int numberOfBytesRead = bufferedInputStream.read(corruptedBytes, 0, logBlockSize);
```

Comment:
if you are buffering already, why do we have to issue such a large read? If you iterator byte-by-byte like before on a buffered reader, do you still have teh same issue?

Reply:
I know there probably is n't seek() to go back and forth.. https://docs.oracle.com/javase/7/docs/api/java/io/BufferedInputStream.html#mark(int) mark should be helpful there to rewind, no?

Reply:
Ping.